### PR TITLE
Add Yolo Format Dataset

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Adding a dataset that support Yolo format dataset, and more.
This does NOT support COCO and Pascal VOC which have a different structure for their labels

### CODE REVIEW

File: documentation/datasets.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Built-in Datasets
+
+## Object Detection
+### Paired Image-Label Dataset
```
Comment (ofrimasad) [inline documentation/datasets.md:4]: Don't be shy; call it YOLO format dataset.
even write explicitly, if you are working with Yolo v1-8, your data is probably organized in this way

File: src/data_gradients/datasets/FolderProcessor.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import os
+import logging
+from typing import List, Optional, Tuple
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_IMG_EXTENSIONS = ("bmp", "dng", "jpeg", "jpg", "mpo", "png", "tif", "tiff", "webp", "pfm")
```
Comment (ofrimasad) [inline src/data_gradients/datasets/FolderProcessor.py:7]: ppm

File: src/data_gradients/datasets/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from data_gradients.datasets.detection import PairedImageLabelDetectionDataset
+from data_gradients.datasets.bdd_dataset import BDDDataset
+
+__all__ = ["PairedImageLabelDetectionDataset", "BDDDataset"]
```
Comment (ofrimasad) [inline src/data_gradients/datasets/__init__.py:4]: call it YoloFormatDetectionDataset

**Review by ofrimasad — COMMENTED** [review]
Looks great. just change the naming and some documentation

File: src/data_gradients/datasets/__init__.py (hunk 3 above)
Comment (Louis-Dupont) [reply src/data_gradients/datasets/__init__.py]: Changed in https://github.com/Deci-AI/data-gradients/pull/114/commits/6f40aaf70aa694e3f8a79dd37ece3155abcbc8ec

File: documentation/datasets.md (hunk 1 above)
Comment (Louis-Dupont) [reply documentation/datasets.md]: Changed in https://github.com/Deci-AI/data-gradients/pull/114/commits/6f40aaf70aa694e3f8a79dd37ece3155abcbc8ec

File: documentation/datasets.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Built-in Datasets
+
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply documentation/datasets.md:2]: Added in https://github.com/Deci-AI/data-gradients/pull/114/commits/6f40aaf70aa694e3f8a79dd37ece3155abcbc8ec

File: src/data_gradients/datasets/FolderProcessor.py (hunk 2 above)
Comment (Louis-Dupont) [reply src/data_gradients/datasets/FolderProcessor.py]: I updated the tuple with all the extensions officially supported by opencv
https://github.com/Deci-AI/data-gradients/pull/114/commits/3ae7a1bf43398ec354b820cae1b9c66e4cd0ec97

We were missing a lot of them, but `dng` and `pfm` are apparently not natively supported with opencv
There are workarounds but do we want to support it? (see https://stackoverflow.com/questions/29487925/opencv-how-to-read-pfm-files)

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**